gulpfile.js

```diff
@@ -0,0 +1,51 @@
+var gulp = require('gulp');
+var gutil = require('gulp-util');
+var ts = require('gulp-typescript');
+var del = require('del');
+var webpack = require('webpack');
+
+gulp.task('clean', function(cb) {
+  del('dist/');
+});
+
+gulp.task('ES5', function() {
+  return gulp.src(['typings/tsd.d.ts', 'src/**/*.ts'])
+    .pipe(ts({
+      target: 'ES5',
+      module: 'commonjs',
+      emitDecoratorMetadata: true,
+      experimentalDecorators: true,
+      declaration: true
+    }))
+    .pipe(gulp.dest('dist/es5'));
+});
+
+gulp.task('ES6', function() {
+  return gulp.src(['typings/tsd.d.ts', 'src/**/*.ts'])
+    .pipe(ts({
+      target: 'ES6',
+      emitDecoratorMetadata: true,
+      experimentalDecorators: true,
+      declaration: true
+    }))
+    .pipe(gulp.dest('dist/es6'));
+});
+
+gulp.task('umd', ['ES5', 'ES6'], function(cb) {
+  webpack({
+    entry: './dist/es5/core.js',
+    output: {
+      filename: 'dist/ng-bootstrap.js',
+      libraryTarget: 'umd'
+    },
+    externals: {
+      "angular2/angular2": "angular2/angular2"
+    }
+  }, function(err, stats) {
+    if (err) throw new gutil.PluginError('webpack', err);
+    gutil.log("[webpack]", stats.toString());
```

Comment:
I took this directly from official docs:

https://webpack.github.io/docs/usage-with-gulp.html